backend/internal/resource/error_constants.go

```diff
@@ -0,0 +1,148 @@
+		ErrorDescription: "Cannot delete resource server/resource that has dependencies",
+	}
+	// ErrorCircularDependency is returned when a circular dependency is detected.
+	ErrorCircularDependency = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1007",
+		Error:            "Circular dependency detected",
+		ErrorDescription: "Setting this parent would create a circular dependency",
+	}
+	// ErrorResourceNotFound is returned when a resource is not found.
+	ErrorResourceNotFound = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1008",
+		Error:            "Resource not found",
+		ErrorDescription: "The resource with the specified id does not exist",
+	}
+	// ErrorActionNotFound is returned when an action is not found.
+	ErrorActionNotFound = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1009",
+		Error:            "Action not found",
+		ErrorDescription: "The action with the specified id does not exist",
+	}
+	// ErrorOrganizationUnitNotFound is returned when organization unit is not found.
+	ErrorOrganizationUnitNotFound = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1010",
+		Error:            "Organization unit not found",
+		ErrorDescription: "The specified organization unit does not exist",
+	}
+	// ErrorInvalidLimit is returned when limit parameter is invalid.
+	ErrorInvalidLimit = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1011",
+		Error:            "Invalid limit parameter",
+		ErrorDescription: "The limit parameter must be a positive integer",
+	}
+	// ErrorInvalidOffset is returned when offset parameter is invalid.
+	ErrorInvalidOffset = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1012",
+		Error:            "Invalid offset parameter",
+		ErrorDescription: "The offset parameter must be a non-negative integer",
+	}
+	// ErrorIdentifierConflict is returned when an identifier already exists.
+	ErrorIdentifierConflict = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1013",
+		Error:            "Identifier conflict",
+		ErrorDescription: "A resource server with the same identifier already exists",
+	}
+	// ErrorHandleConflict is returned when a handle already exists.
+	ErrorHandleConflict = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "RES-1014",
+		Error:            "Handle conflict",
+		ErrorDescription: "The same handle already exists within the specified resource",
+	}
+)
+
+// Internal error constants.
+var (
+	// ErrResourceServerNotFound is returned when the resource server is not found.
+	ErrResourceServerNotFound = errors.New("resource server not found")
```

Comment:
These errors are only used internally to communicate errors between store and service layer right?. Shall we make them private?